packages/react-native/src/theme/types.ts

```diff
@@ -0,0 +1,38 @@
+import { PartialDeep } from 'type-fest';
+
+import { Tokens } from './tokens';
+
+type Override = Omit<Theme, 'overrides'>;
+
+export type ColorMode = 'light' | 'dark' | 'system';
+
+/**
+ * A Theme just needs a name.
+ * Users can define any tokens they need, but they don't need a complete theme with all tokens.
+ */
+export interface Theme {
+  colorMode?: ColorMode;
+  /**
+   * The name of the theme.
+   */
+  name: string;
+  tokens?: PartialDeep<Tokens>;
```

Comment:
Rather than using `PartialDeep`, let's build the `Theme` that we export publicly with all keys as optional by default, and then use that interface to create a strict/required theme.